src/server/hiring_process.go

```diff
@@ -208,3 +220,76 @@ func VacancyTable(
 		c.JSON(http.StatusOK, vacancies)
 	}
 }
+
+// ListDepartments godoc
+// @Summary List departments
+// @Schemes
+// @Description Return a list of departments with id and title
+// @Tags departments
+// @Produce json
+// @Success 200 {array} model.Suggestion
+// @Router /departments [get]
+func ListDepartments(
+	client *ent.Client,
+) func(c *gin.Context) {
+	return func(c *gin.Context) {
+		departments, err := service.ListDepartments(c, client)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{
+				"error":   "Failed to list departments",
+				"details": err.Error(),
+			})
+			return
+		}
+		c.JSON(http.StatusOK, departments)
+	}
+}
+
+// CreateGroupAcess godoc
+// @Summary Create a new group access
+// @Schemes
+// @Description Create a new group access with name and related departments
+// @Tags group_acess
+// @Accept json
+// @Produce json
+// @Param body body service.CreateGroupAcessRequest true "Group Access Info"
+// @Success 201 {object} ent.GroupAcess
+// @Router /group_acess [post]
+func CreateGroupAcess(client *ent.Client) func(c *gin.Context) {
+	return func(c *gin.Context) {
+		var request service.CreateGroupAcessRequest
+
+		if err := c.ShouldBindJSON(&request); err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request body"})
+			return
+		}
+
+		group, err := service.CreateGroupAcess(c, client, request)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+			return
+		}
+
+		c.JSON(http.StatusCreated, group)
+	}
+}
+
+// ListGroupAcess godoc
+// @Summary List group access with departments
+// @Schemes
+// @Description Return a list of group access with id, name, and departments
+// @Tags group_acess
+// @Produce json
+// @Success 200 {array} service.GroupAcessReturn
+// @Router /group_acess [get]
```

Comment:
essa rota do swagger não condiz com a rota que está sendo criada, o certo seria:

```suggestion
// @Router /group_access/List [get]
```

ou, também padronizando para snake_case:

```suggestion
// @Router /group_access/list [get]
```

ou, tirando a necessidade do subpath `list`:

```suggestion
// @Router /group_access [get]
```

> [!NOTE]
segunda e terceira sugestão envolvem trocar na linha 41 tbm

> [!NOTE]
no caso da terceira sugestão, na linha 41 ficaria `groupAccess.GET("", ListGroupAcess(dbClient))` (string vazia como `path`)